cmd/outline-ss-server/main.go

```diff
@@ -297,6 +340,55 @@ func (s *OutlineServer) runConfig(config Config) (func() error, error) {
 						go service.PacketServe(pc, func(ctx context.Context, conn net.Conn) {
 							associationHandler.HandleAssociation(ctx, conn, s.serviceMetrics.AddOpenUDPAssociation(conn))
 						}, s.serverMetrics)
+					} else if cfg.WebsocketStream != nil {
+						if _, exists := webServers[cfg.WebsocketStream.WebServer]; !exists {
+							return fmt.Errorf("listener references unknown web server `%s`", cfg.WebsocketStream.WebServer)
+						}
+						mux := webServers[cfg.WebsocketStream.WebServer]
+						// TODO: Support a "half-closed" state for WebSockets.
+						handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+							handler := func(wsConn *websocket.Conn) {
+								defer wsConn.Close()
+								ctx, contextCancel := context.WithCancel(context.Background())
+								defer contextCancel()
+								raddr, err := net.ResolveTCPAddr("tcp", r.RemoteAddr)
+								if err != nil {
+									slog.Error("failed to upgrade", "err", err)
+									w.WriteHeader(http.StatusBadGateway)
+									return
+								}
+								conn := &streamConn{&wrappedConn{Conn: wsConn, raddr: raddr}}
```

Comment:
Need a better name than `wrappedConn`. It doesn't convey any meaning.
Maybe `replaceAddrConn`?